### PR TITLE
Create withdrawal checker class

### PR DESCRIPTION
**Pull request by tyuyang**
First step in creating withdrawal limit. This is going to take some time

### CODE REVIEW

File: src/main/java/seedu/bankwithus/Account.java [hunk 1]
```diff
@@ -16,6 +18,15 @@ public class Account {
...
         this.balance = balance;
+        this.withdrawalChecker = new WithdrawalChecker();
+    }
+
+    //@@author tyuyang
+    public Account(String name, String balance, String totalAmtWithdrawn,
+            LocalDate lastWithdrawnDate) {
+        this.name = name;
+        this.balance = balance;
+        this.withdrawalChecker = new WithdrawalChecker(totalAmtWithdrawn, lastWithdrawnDate);
```
Comment (tyuyang) [inline src/main/java/seedu/bankwithus/Account.java:29]: Created new constructor and modified old constructor to instantiate withdrawalChecker

File: src/main/java/seedu/bankwithus/Account.java [hunk 2]
```diff
@@ -36,5 +47,8 @@ public void subtractBalance(float currentBalance, float withdrawal) {
         DecimalFormat df = new DecimalFormat("#.##");
         String formatted = df.format(currentBalance - withdrawal);
         this.balance = String.valueOf(formatted);
+
+        withdrawalChecker.updateTotalAmtWithdrawn(withdrawal);
```
Comment (tyuyang) [inline src/main/java/seedu/bankwithus/Account.java:51]: Called function to update withdrawals this month in the subtractBalance function

File: src/main/java/seedu/bankwithus/WithdrawalChecker.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+public class WithdrawalChecker {
+    private String totalAmtWithdrawn;
+    private LocalDate lastWithdrawnDate;
+
+    /**
+     * Instantiates withdrawalChecker without any amount withdrawn
+     */
+    public WithdrawalChecker() {
+        this.totalAmtWithdrawn = "0";
+    }
```
Comment (tyuyang) [inline src/main/java/seedu/bankwithus/WithdrawalChecker.java:15]: Default constructor if account is new or does not have any withdrawals yet

File: src/main/java/seedu/bankwithus/WithdrawalChecker.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    /**
+     * Instantiates withdrawalChecker with amount withdrawn and 
+     * last known withdrawal date
+     * @param totalAmtWithdrawn amount withdrawn
+     * @param lastWithdrawnDate last known withdrawal date
+     */
+    public WithdrawalChecker(String totalAmtWithdrawn, LocalDate lastWithdrawnDate) {
+        this.totalAmtWithdrawn = totalAmtWithdrawn;
+        this.lastWithdrawnDate = lastWithdrawnDate;
+    }
```
Comment (tyuyang) [inline src/main/java/seedu/bankwithus/WithdrawalChecker.java:26]: Constructor for accounts that have history of withdrawals

File: src/main/java/seedu/bankwithus/WithdrawalChecker.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+     */
+    public void updateTotalAmtWithdrawn(float withdrawal) {
+        LocalDate currentDate = LocalDate.now();
+        DecimalFormat df = new DecimalFormat("#.##");
+        if (lastWithdrawnDate == null) {
+            lastWithdrawnDate = currentDate;
+            String formatted = df.format(withdrawal);
+            totalAmtWithdrawn = String.valueOf(formatted);
+            return;
+        }
```
Comment (tyuyang) [inline src/main/java/seedu/bankwithus/WithdrawalChecker.java:48]: If new account or no withdrawal history

File: src/main/java/seedu/bankwithus/WithdrawalChecker.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        assert totalAmtWithdrawn != null;
+        if (lastWithdrawnDate.getMonth() == currentDate.getMonth() && 
+                lastWithdrawnDate.getYear() == currentDate.getYear()) {
+            String formatted = df.format(Float.parseFloat(totalAmtWithdrawn) + withdrawal);
+            totalAmtWithdrawn = String.valueOf(formatted);
+        } else {
```
Comment (tyuyang) [inline src/main/java/seedu/bankwithus/WithdrawalChecker.java:54]: If same month and year, add to total amt withdrawn

File: src/main/java/seedu/bankwithus/WithdrawalChecker.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+            String formatted = df.format(withdrawal);
+            totalAmtWithdrawn = String.valueOf(formatted);
+        }
```
Comment (tyuyang) [inline src/main/java/seedu/bankwithus/WithdrawalChecker.java:57]: Else reset withdrawal amount

File: src/main/java/seedu/bankwithus/WithdrawalChecker.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        lastWithdrawnDate = currentDate;
```
Comment (tyuyang) [inline src/main/java/seedu/bankwithus/WithdrawalChecker.java:58]: Update lastWithdrawn date for tracking purposes

**Review by xiaoge26 — APPROVED** [review]
Looks good to me!

**Pull request merged**